Buffer detection

I'm trying to build this module with webpack in [js-ipfs/examples/browser-webpack](https://github.com/ipfs/js-ipfs/tree/master/examples/browser-webpack) and the Buffer detection doesn't work:

```
ERROR in ../../node_modules/cborg/esm/lib/byte-utils.js 1:32-39
Module not found: Error: Can't resolve 'process/browser' in '/Users/alex/Documents/Workspaces/ipfs/js-ipfs/node_modules/cborg/esm/lib'
Did you mean 'browser.js'?
BREAKING CHANGE: The request 'process/browser' failed to resolve only because it was resolved as fully specified
(probably because the origin is a '*.mjs' file or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
 @ ../../node_modules/cborg/esm/lib/encode.js 9:0-40 192:13-18
 @ ../../node_modules/cborg/esm/cborg.js 1:0-41 7:0-12:2
 @ ../../packages/ipfs-core/src/components/pin/pin-manager.js
 @ ../../packages/ipfs-core/src/components/index.js 93:20-48
 @ ../../packages/ipfs-core/src/index.js 27:4-27
 @ ../../packages/ipfs/src/index.js 12:35-55
 @ ./src/components/app.js 11:11-26
 @ ./src/components/index.js 7:10-26
```

It's good to use buffer because `Buffer.allocUnsafe(size)` is significantly faster than `new Uint8Array(size)`, but it's bad to use node globals.

Solutions:

1. Use the [process](https://www.npmjs.com/package/process) module:

```js
import process from 'process'

export const useBuffer = !process.browser && global.Buffer && typeof global.Buffer.isBuffer === 'function';
```

2. Use the [buffer](https://www.npmjs.com/package/buffer) module:

```js
import { Buffer } from 'buffer'

export const useBuffer = true;

// replace global.Buffer with Buffer
```

--- a/lib/byte-utils.js
+++ b/lib/byte-utils.js
@@ -1,16 +1,16 @@
 // Use Uint8Array directly in the browser, use Buffer in Node.js but don't
 // speak its name directly to avoid bundlers pulling in the `Buffer` polyfill
 
-export const useBuffer = global.process &&
-  !global.process.browser &&
-  global.Buffer &&
-  typeof global.Buffer.isBuffer === 'function'
+export const useBuffer = globalThis.process &&
+  !globalThis.process.browser &&
+  globalThis.Buffer &&
+  typeof globalThis.Buffer.isBuffer === 'function'
 
 const textDecoder = new TextDecoder()
 const textEncoder = new TextEncoder()
 
 function isBuffer (buf) {
-  return useBuffer && global.Buffer.isBuffer(buf)
+  return useBuffer && globalThis.Buffer.isBuffer(buf)
 }
 
 export function asU8A (buf) {
@@ -24,7 +24,7 @@ export function asU8A (buf) {
 export const toString = useBuffer
   ? (bytes, start, end) => {
       return end - start > 64
-        ? global.Buffer.from(bytes.subarray(start, end)).toString('utf8')
+        ? globalThis.Buffer.from(bytes.subarray(start, end)).toString('utf8')
         : utf8Slice(bytes, start, end)
     }
   /* c8 ignore next 5 */
@@ -36,7 +36,7 @@ export const toString = useBuffer
 
 export const fromString = useBuffer
   ? (string) => {
-      return string.length > 64 ? global.Buffer.from(string) : utf8ToBytes(string)
+      return string.length > 64 ? globalThis.Buffer.from(string) : utf8ToBytes(string)
     }
   /* c8 ignore next 3 */
   : (string) => {
@@ -64,8 +64,8 @@ export const concat = useBuffer
   ? (chunks, length) => {
       // might get a stray plain Array here
       /* c8 ignore next 1 */
-      chunks = chunks.map((c) => c instanceof Uint8Array ? c : global.Buffer.from(c))
-      return asU8A(global.Buffer.concat(chunks, length))
+      chunks = chunks.map((c) => c instanceof Uint8Array ? c : globalThis.Buffer.from(c))
+      return asU8A(globalThis.Buffer.concat(chunks, length))
     }
   /* c8 ignore next 13 */
   : (chunks, length) => {
@@ -85,7 +85,7 @@ export const concat = useBuffer
 export const alloc = useBuffer
   ? (size) => {
       // we always write over the contents we expose so this should be safe
-      return global.Buffer.allocUnsafe(size)
+      return globalThis.Buffer.allocUnsafe(size)
     }
   /* c8 ignore next 3 */
   : (size) => {
@@ -97,7 +97,7 @@ export const toHex = useBuffer
       if (typeof d === 'string') {
         return d
       }
-      return global.Buffer.from(toBytes(d)).toString('hex')
+      return globalThis.Buffer.from(toBytes(d)).toString('hex')
     }
   /* c8 ignore next 6 */
   : (d) => {
@@ -112,7 +112,7 @@ export const fromHex = useBuffer
       if (hex instanceof Uint8Array) {
         return hex
       }
-      return global.Buffer.from(hex, 'hex')
+      return globalThis.Buffer.from(hex, 'hex')
     }
   /* c8 ignore next 12 */
   : (hex) => {

